### ISSUE TITLE
My changes are auto-merged away

### ISSUE THREAD

**Issue by enikao** [IETS3/iets3.opensource#1186]
Am I doing something wrong there? It's _very_ annoying to debug this stuff several times.

Compare e.g. #1119 with `maintenance/mps20232`:

#1119:
```
<solution name="org.iets3.core.expr.math.interpreter" uuid="b804a851-ecf0-4ad4-a0af-ae720b39191a" moduleVersion="0">
  <models>
    <modelRoot type="java_classes" contentPath="${module}/lib">
      <sourceRoot location="big-math.jar" />
    </modelRoot>
    <modelRoot type="default" contentPath="${module}">
      <sourceRoot location="models" />
    </modelRoot>
  </models>
```

vs. 

mps20232:
```
<solution name="org.iets3.core.expr.math.interpreter" uuid="b804a851-ecf0-4ad4-a0af-ae720b39191a" moduleVersion="0">
  <models>
    <modelRoot contentPath="${module}" type="default">
      <sourceRoot location="models" />
      <sourceRoot location="lib" />
    </modelRoot>
    <modelRoot contentPath="${module}/lib" type="java_classes">
      <sourceRoot location="big-math.jar" />
    </modelRoot>
  </models>
```

**Comment by alexanderpann** [comment]
I investigated the old issues that we had with your lost changes. The ones related to the exceptions from the tracer actions were indeed caused by re-runnable MPS migrations, I fixed this issue and also two other instances in the codebase. Another PR got lost where I still don't know what happened, but we restored that one. With https://github.com/IETS3/iets3.opensource/pull/1119 it was very unfortunate timing: On the same day we migrated to 2024.1  so your PR got merged into master which was already 2024.1 at that time. We can of course bring back that change to 2023.2.
